### ISSUE TITLE
Tracking: additional layout options, design tools and refinements

### ISSUE THREAD

**Issue by andrewserong** [WordPress/gutenberg#44720]
This supersedes https://github.com/WordPress/gutenberg/issues/39336 for tracking upcoming tasks around the layout block support.

As of WordPress 6.1, the layout block support [has been refactored](https://github.com/WordPress/gutenberg/pull/40875) to generate styles based on a central set of layout definitions, [a new constrained layout type](https://github.com/WordPress/gutenberg/pull/42763) has been added, along with [root padding aware alignments](https://github.com/WordPress/gutenberg/pull/42085). This issue tracks additional tasks and progress slated for WordPress 6.2 and beyond.

## Additional layout options and design tools

* [x] https://github.com/WordPress/gutenberg/issues/44467
* [x] https://github.com/WordPress/gutenberg/issues/45826 (related to above)
* [ ] https://github.com/WordPress/gutenberg/issues/42385
* [x] https://github.com/WordPress/gutenberg/issues/30121
* [x] https://github.com/WordPress/gutenberg/issues/45117 (related to above)
* [x] https://github.com/WordPress/gutenberg/issues/43433
* [x] https://github.com/WordPress/gutenberg/issues/47809

## Improve and extend existing layout support

* [ ] https://github.com/WordPress/gutenberg/issues/44560
* [ ] https://github.com/WordPress/gutenberg/issues/43248
* [x] https://github.com/WordPress/gutenberg/issues/43140
* [x] https://github.com/WordPress/gutenberg/issues/44557
* [x] https://github.com/WordPress/gutenberg/issues/35849
* [x] https://github.com/WordPress/gutenberg/issues/45868 (related to above)
* [x] https://github.com/WordPress/gutenberg/issues/44899
* [x] https://github.com/WordPress/gutenberg/pull/44600
* [x] https://github.com/WordPress/gutenberg/issues/43404
* [ ] https://github.com/WordPress/gutenberg/issues/47807

## Nice-to-haves

* [ ] https://github.com/WordPress/gutenberg/issues/37462

## Bugs, tasks, and issues not covered here

This issue is not an exhaustive list of all layout related issues, tasks, or bugs. For bug reports and smaller tasks, we can use the [`[Feature] Layout`](https://github.com/WordPress/gutenberg/issues?q=is%3Aopen+is%3Aissue+label%3A%22%5BFeature%5D+Layout%22) issue label for tracking and visibility.

**Comment by andrewserong** [comment]
Update 2022/10/26:

* https://github.com/WordPress/gutenberg/pull/44600 — this PR has now been merged, which moves layout classnames and styles to be attached to the inner blocks wrapper rather than the block outer wrapper. This was an early wish list for the layout block support and unblocks opting blocks with nested markup in to using the layout block support (most notably the cover block). Kudos @tellthemachines for landing this!
* [Fixed position header and footer](https://github.com/WordPress/gutenberg/issues/30121) is being explored, along with a WIP PR in https://github.com/WordPress/gutenberg/pull/44723 — there will likely be lots of edge cases in attempting to implement this, so the plan right now is to get the PR testable so that we can see what those edge cases are
* https://github.com/WordPress/gutenberg/issues/45117 — this is an interesting requirement for laying out post or page templates. A potential solution is to use minimum height on the post content block, which is slightly outside of the layout block support, but is being explored (again) in https://github.com/WordPress/gutenberg/pull/45300

**Comment by andrewserong** [comment]
Update for 2022/11/16:

* There is now a variation picker UI in the Group block placeholder state (https://github.com/WordPress/gutenberg/pull/43496)
* The minimum height block support has been merged, and is now opted-in for both the Group and Post Content blocks. There is a follow-up issue for tweaks to the UI (https://github.com/WordPress/gutenberg/issues/45501), as well as feedback that there are other blocks that could be opted-in too (e.g. Column block). A reminder that dimensions opt-ins are tracked in #43243.
* There is a WIP PR under review for adding width / size controls to child blocks of flex layouts over in: https://github.com/WordPress/gutenberg/pull/45364
* There has been useful design feedback on the WIP sticky position support PR (https://github.com/WordPress/gutenberg/pull/44723). The next step for this one is to first land a refactor to the layout panel to use the ToolsPanel component (as raised in https://github.com/WordPress/gutenberg/issues/44560). I'll dig into that task this week.

**Comment by andrewserong** [comment]
Update for 2022/12/07:

* Fit/Fill/Fixed size controls have been added to child blocks of Row and Stack now, merged in: https://github.com/WordPress/gutenberg/pull/45364
* For consistency with other UI controls and to make it easier to make adjustments, a slider has been added alongside the Min Height control: https://github.com/WordPress/gutenberg/pull/45875
* Work continues on exploring fixed/sticky support for container blocks in: https://github.com/WordPress/gutenberg/pull/46142. The current exploration now decouples the concept from Layout, as there are potential future use cases for _non_ container blocks (e.g. one day being able to set relative/absolute, to position images for example). There's still more discussion/work on this topic and a few blockers to resolve, so updates will happen over in the tracking issue for that feature here: https://github.com/WordPress/gutenberg/issues/30121#issuecomment-1330133950
* A PR has been merged that improves the server-rendering performance of layout by reducing calls to get global settings: https://github.com/WordPress/gutenberg/pull/46074

**Comment by andrewserong** [comment]
Update for 2023/02/03:

Since the last update, the following changes have been merged and appear to be slated for inclusion in 6.2:

* Fixes for layout / block spacing output in multi site instances (https://github.com/WordPress/gutenberg/pull/46388 and  https://github.com/WordPress/gutenberg/pull/46712)
* Additional controls for flex layouts (Stretch to Fill for vertical alignment on Row block, and Stretch and Space between options for Stack block): https://github.com/WordPress/gutenberg/pull/47134
* A stability and code quality improvement for how layout is passed to inner blocks, which forms one part of overall stabilising the layout block support: https://github.com/WordPress/gutenberg/pull/47477
* Add layout child sizing controls to the Navigation block: https://github.com/WordPress/gutenberg/pull/47584

Adjacent to the layout support, the sticky position block support has landed, with support for Group blocks at the root of the document. The current tracking issue for position support is over at: https://github.com/WordPress/gutenberg/issues/47043

Work continues on exploring a fix for conflicts between margins set at the block level in global styles, and layout styles output (https://github.com/WordPress/gutenberg/issues/43404) over in https://github.com/WordPress/gutenberg/pull/47399 — the fix is tricky to land on due to the subtleties of the styles at play, but the hope is to see if a fix can land in time for 6.2.

Kudos @tellthemachines for all the recent progress on Layout PRs!

**Comment by talldan** [comment]
Should layout be added to the Media & Text block? It's an interesting one.

Using block gap is something that was raised in this issue - https://github.com/WordPress/gutenberg/issues/49406.

**Comment by andrewserong** [comment]
> Should layout be added to the Media & Text block? It's an interesting one.

Thanks for raising that issue! I've left a comment — I tend to agree that it'd be more about arranging the inner blocks. If it were to be opted in, I wonder if it might use a similar approach as @tellthemachines used to convert the Cover block in https://github.com/WordPress/gutenberg/pull/45326 🤔

**Comment by andrewserong** [comment]
Now that the layout block support has been stabilised within WP core for a couple of releases (6.3 and 6.4, and from this PR: https://github.com/WordPress/gutenberg/pull/51434 onwards), I'll close out this tracking issue now, as its goal was largely to get the block support to that state.

There are still many enhancements and fixes to be made for the layout block support (such as simplifying layout in https://github.com/WordPress/gutenberg/issues/42385), but they can be explored in follow up issues via the [`[Feature] Layout` label](https://github.com/WordPress/gutenberg/labels/%5BFeature%5D%20Layout) or subsequent tracking issues.
